utils/indexed_datasets.py

```diff
@@ -56,15 +55,16 @@ def add_item(self, item):
                 k: item.get(k)
                 for k in self.allowed_attr
             }
-        s = pickle.dumps(item)
-        n_bytes = self.out_file.write(s)
-        self.byte_offsets.append(self.byte_offsets[-1] + n_bytes)
+        with self.lock:
+            item_no = self.counter
+            self.counter += 1
+        for k, v in item.items():
```

Comment:
Maybe better to change line 53-57 like this?
```python
item = {
    k: item[k]
    for k in self.allowed_attr
    if k in item
}
```

Reply:
Also, is it necessary to acquire a lock here? Will there be cases where multiple processes writing items at the same time?

Reply:
Temporarily disable locking